state/redis/redis.go

```diff
@@ -344,43 +347,40 @@ func (r *StateStore) Set(ctx context.Context, req *state.SetRequest) error {
 		firstWrite = 0
 	}
 
-	var bt []byte
-	var setQuery string
-	if contentType, ok := req.Metadata[daprmetadata.ContentType]; ok && contentType == contenttype.JSONContentType && rediscomponent.ClientHasJSONSupport(r.client) {
-		setQuery = setJSONQuery
-		bt, _ = utils.Marshal(&jsonEntry{Data: req.Value}, r.json.Marshal)
+	if req.Metadata[daprmetadata.ContentType] == contenttype.JSONContentType && r.clientHasJSON {
+		bt, _ := utils.Marshal(jsonEntry{Data: req.Value}, r.json.Marshal)
```

Comment:
```suggestion
		bt, _ := utils.Marshal(&jsonEntry{Data: req.Value}, r.json.Marshal)
```
why was pointer ref removed from older code?